ekncontent/ekncontent/eknc-content-object-model.c

```diff
@@ -0,0 +1,580 @@
+  /**
+   * EkncContentObjectModel:original-uri:
+   *
+   * URI where the original version of this content can be downloaded
+   *
+   * This property is distinct from source-uri, which represents the URI
+   * where the article was downloaded from during database build.
+   *
+   * However, on an #ArticleObjectModel with source equal to
+   * "wikipedia", "wikihow", "wikisource", or "wikibooks", it will be
+   * set to the value of source-uri if it is not present in the
+   * database, for backwards compatibility reasons.
+   */
+  eknc_content_object_model_props[PROP_ORIGINAL_URI] =
+    g_param_spec_string ("original-uri", "Original URI",
+      "URI where the original version of this content can be downloaded",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:thumbnail-uri:
+   *
+   * URI of the the thumbnail image
+   *
+   * Usually the EKN ID of an #ImageObjectModel.
+   */
+  eknc_content_object_model_props[PROP_THUMBNAIL_URI] =
+    g_param_spec_string ("thumbnail-uri", "Thumbnail URI",
+       "URI of the thumbnail image",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:language:
+   *
+   * The language for this content object. Defaults to an empty string.
+   */
+  eknc_content_object_model_props[PROP_LANGUAGE] =
+    g_param_spec_string ("language", "Language",
+      "Language of the document or media object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:copyright-holder:
+   *
+   * The copyright holder for this content object. Defaults to an empty string.
+   */
+  eknc_content_object_model_props[PROP_COPYRIGHT_HOLDER] =
+    g_param_spec_string ("copyright-holder", "Copyright Holder",
+      "The copyright holder of the object",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:source-uri:
+   *
+   * URI where this content was downloaded from during database build
+   *
+   * TODO Should we check to always have a value for source-uri?
+   * Don"t use this property for user-visible things.
+   * It is only used internally.
+   */
+  eknc_content_object_model_props[PROP_SOURCE_URI] =
+    g_param_spec_string ("source-uri", "Source URL",
+      "URI where this content was downloaded from during database build",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:content-type:
+   *
+   * The source content"s mimetype
+   */
+  eknc_content_object_model_props[PROP_CONTENT_TYPE] =
+    g_param_spec_string ("content-type", "Object Content Type",
+      "Mimetype of the source content",
+      "", G_PARAM_READWRITE | G_PARAM_CONSTRUCT_ONLY | G_PARAM_STATIC_STRINGS);
+  /**
+   * EkncContentObjectModel:synopsis:
+   *
+   * The synopsis for this content object. Defaults to an empty string.
```

Comment:
nitpick: As long as you're touching this code, get rid of the "Defaults to ..." comments; the default values are automatically extracted by gtk-doc.